plugin/distribution/pull.go

```diff
@@ -143,8 +143,7 @@ func Pull(name string, rs registry.Service, metaheader http.Header, authConfig *
 		logrus.Debugf("pull.go: error in json.Unmarshal(): %v", err)
 		return nil, err
 	}
-	if m.Config.MediaType != MediaTypeConfig &&
-		m.Config.MediaType != "application/vnd.docker.plugin.image.v0+json" {
+	if m.Config.MediaType != MediaTypeConfig {
```

Comment:
Yes, `application/vnd.docker.plugin.image.v0+json` was a placeholder for the final type that 1.13 would ship with. 
If 1.12.x encounters a plugin with mediatype `application/vnd.docker.plugin.v1+json`, it implies that the plugin was created using 1.13+. Hence it is not supported in 1.12.x